contracts/utils/cryptography/ECDSA.sol

```diff
@@ -43,6 +43,13 @@ library ECDSA {
      * this function rejects them by requiring the `s` value to be in the lower
      * half order, and the `v` value to be either 27 or 28.
      *
+     * NOTE: This function also prevents signature malleability by only supporting 65 bytes long signatures, and
+     * rejecting eip-2098 short signatures. While this guarantee is still present, it is DEPRECATED and will be
+     * removed in the next major release (v6.0). Developers SHOULD NOT use signatures as unique identifiers. If an
+     * operation must me marked as consumed to prevent replayability, either the `hash` (or the `hash`/`recovered`
+     * pair if multiple accounts are to sign the same hash) should be invalidated. Nonces are also a viable solution.
+     * Marking the signatures as consumed is very strongly discouraged.
```

Comment:
This text is a little too long for my taste but it's ok.

The deprecation needs to be added to the changelog as well.

```suggestion
     * NOTE: This function protects against malleability by only supporting 65 bytes long signatures, and
     * rejecting EIP-2098 short signatures. This guarantee is DEPRECATED and will be
     * removed in the next major release (v6.0). Developers SHOULD NOT use signatures as unique identifiers. If an
     * operation must be marked as consumed to prevent replayability, either the `hash` (or the `hash`/`recovered`
     * pair if multiple accounts are to sign the same hash) should be invalidated. Nonces are also a viable solution.
```